lib/segment/src/types.rs

```diff
@@ -1196,6 +1196,32 @@ impl GeoRadius {
     }
 }
 
+/// Geo filter request
+///
+/// Matches coordinates inside the polygon, defined by the given coordinates in order.
+#[derive(Debug, Deserialize, Serialize, JsonSchema, Clone, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub struct GeoPolygon {
+    /// Ordered list of coordinates representing the vertices of a polygon.
+    pub points: Vec<GeoPoint>,
+}
+
+impl GeoPolygon {
+    pub fn check_point(&self, lon: f64, lat: f64) -> bool {
+        let polygon_points: Vec<Coord<f64>> = self
+            .points
+            .iter()
+            .map(|p| Coord { x: p.lon, y: p.lat })
+            .collect();
+
+        let polygon = Polygon::new(polygon_points.into(), vec![]);
```

Comment:
for other reviewers, the second argument here is a list of point to form an inner polygon (exterior + interior = donut like shape).